webapp/src/components/AssetTopbar/SearchBarDropdown/SearchBarDropdown.spec.tsx

```diff
@@ -0,0 +1,532 @@
+    category: NFTCategory.WEARABLE,
+    onSearch: jest.fn(),
+    fetchedCreators: [],
+    isLoadingCreators: false,
+    onFetchCreators: jest.fn(),
+    onClickOutside: jest.fn()
+  }
+  return renderWithProviders(
+    <ResponsiveContext.Provider value={{ width: 900 }}>
+      <SearchBarDropdown {...defaultProps} {...props} />
+    </ResponsiveContext.Provider>
+  )
+}
+
+describe('SearchBarDropdown', () => {
+  describe('when providing a search term', () => {
+    let props: Partial<SearchBarDropdownProps>
+    beforeEach(() => {
+      props = {
+        searchTerm: 'example'
+      }
+    })
+
+    describe('and its still loading', () => {
+      beforeEach(() => {
+        ;(catalogAPI.get as jest.Mock).mockResolvedValueOnce({
+          data: []
+        })
+      })
+      it('should render the component', () => {
+        const { getByTestId } = renderSearchDropBarDropdown(props)
+        expect(getByTestId('search-bar-dropdown')).toBeInTheDocument()
+      })
+
+      it('should render search tabs correctly', () => {
+        const { container } = renderSearchDropBarDropdown(props)
+        expect(container.querySelector('.dcl.tabs')).toBeInTheDocument()
+      })
+
+      it('renders loading skeleton while loading', () => {
+        const { getAllByTestId } = renderSearchDropBarDropdown(props)
+        expect(getAllByTestId(SKELETONS_DATA_TEST_ID)).toHaveLength(5)
+      })
+    })
+
+    describe('and the user is searching for collectibles', () => {
+      describe('and there are results', () => {
+        beforeEach(() => {
+          ;(catalogAPI.get as jest.Mock).mockResolvedValueOnce({
+            data: [MOCKED_ITEM]
+          })
+        })
+
+        it('should render the results', async () => {
+          const { getByText } = renderSearchDropBarDropdown(props)
+          await waitFor(() => {
+            expect(getByText(MOCKED_ITEM.name)).toBeInTheDocument()
+          })
+        })
+
+        it('should have the correct asset url with in the result rendered', async () => {
+          const onSearch = jest.fn()
+          const { container } = renderSearchDropBarDropdown({
+            ...props,
+            onSearch
+          })
+          await waitFor(async () => {
+            expect(within(container).getByRole('link')).toHaveAttribute(
+              'href',
+              getAssetUrl(MOCKED_ITEM)
+            )
+          })
+        })
+
+        it('should have the save the result in the recent searches on the click event', async () => {
+          const onSearch = jest.fn()
+          const { container } = renderSearchDropBarDropdown({
+            ...props,
+            onSearch
+          })
+          await waitFor(async () => {
+            const link = within(container).getByRole('link')
+            await fireEvent.click(link)
+            expect(
+              localStorage.getItem(LOCAL_STORAGE_RECENT_SEARCHES_KEY)
+            ).toBe(JSON.stringify([MOCKED_ITEM]))
+          })
+        })
```

Comment:
Would you mind writing this test into a describe?
```suggestion
        describe("when clicking the component", () => {
          it('should have the save the result in the recent searches', async () => {
            const onSearch = jest.fn()
            const { container } = renderSearchDropBarDropdown({
              ...props,
              onSearch
            })
            await waitFor(async () => {
              const link = within(container).getByRole('link')
              await fireEvent.click(link)
              expect(
                localStorage.getItem(LOCAL_STORAGE_RECENT_SEARCHES_KEY)
              ).toBe(JSON.stringify([MOCKED_ITEM]))
            })
          })
        })
```
The same would apply to other tests that perform actions or describe contexts of actions.